src/rust/engine/fs/src/store.rs

```diff
@@ -106,7 +109,8 @@ impl Store {
   }
 
   ///
-  /// Loads the bytes of the file with the passed fingerprint, and returns the result of applying f
+  /// Loads the bytes of the file with the passed fingerprint from the local store
+  /// and back-fill from remote when necessary, and returns the result of applying f
```

Comment:
The back-filling comments here are interesting... It will only happen with `self.remote` is not `None`... I'm not sure where a good place to document the general "will backfill if remote is set" behaviour would be - maybe just saying so here?